pkg/destroy/aws/aws.go

```diff
@@ -185,19 +185,25 @@ func (o *ClusterUninstaller) RunWithContext(ctx context.Context) ([]string, erro
 	// running on the cluster creating new resources while we are attempting to delete resources, which could leak
 	// the new resources.
 	ec2Client := ec2.New(awsSession)
+	shutdownTimeout := time.Now()
 	err = wait.PollImmediateUntil(
 		time.Second*10,
 		func() (done bool, err error) {
-			instancesToDelete, err := o.findEC2Instances(ctx, ec2Client, deleted)
+			instancesRunning, instancesNotTerminated, err := o.findEC2Instances(ctx, ec2Client, deleted)
 			if err != nil {
 				o.Logger.WithError(err).Info("error while finding EC2 instances to delete")
 				if err := ctx.Err(); err != nil {
 					return false, err
 				}
 			}
-			if len(instancesToDelete) == 0 && err == nil {
+			if len(instancesNotTerminated) == 0 && len(instancesRunning) == 0 && err == nil {
 				return true, nil
 			}
+			instancesToDelete := instancesRunning
+			if time.Since(shutdownTimeout) > 5*time.Second {
```

Comment:
I assume that 5 seconds is just for your initial testing and that the actual delay between mass terminates will be much longer.